devbox.go

```diff
@@ -238,6 +238,50 @@ func (d *Devbox) Shell() error {
 	return shell.Run(nixShellFilePath)
 }
 
+// TODO: consider unifying the implementations of RunScript and Shell.
+func (d *Devbox) RunScript(scriptName string) error {
+	if err := d.ensurePackagesAreInstalled(install); err != nil {
+		return err
+	}
+
+	plan := d.ShellPlan()
+
+	profileDir, err := d.profileDir()
+	if err != nil {
+		return err
+	}
+
+	nixShellFilePath := filepath.Join(d.srcDir, ".devbox/gen/shell.nix")
+	script := d.cfg.Shell.Scripts[scriptName]
+	if script == nil {
+		return errors.Errorf("unable to find a script with name %s", scriptName)
+	}
+
+	shell, err := nix.DetectShell(
+		nix.WithPlanInitHook(strings.Join(plan.ShellInitHook, "\n")),
+		nix.WithProfile(profileDir),
+		nix.WithHistoryFile(filepath.Join(d.srcDir, shellHistoryFile)),
+		nix.WithUserScript(scriptName, script.String()))
+
+	if err != nil {
+		fmt.Print(err)
+		shell = &nix.Shell{}
+	}
+
+	shell.UserInitHook = d.cfg.Shell.InitHook.String()
+	return shell.Run(nixShellFilePath)
```

Comment:
oh boyo boy. I thought this would reuse the implementation of `box.Exec`....
but I agree with your implementation more (as in, we should strive to emulate the `devbox shell` environment which `box.Exec` doesn't seem to).

So, there are three different implementations now:
- this `RunScript`
- `box.Exec`
- the `box.Shell`  (except this doesn't run a specific command, but starts an interactive shell)

I feel:
1. `box.Exec` and `RunScript` should use the exact same implementation
2. `box.Shell` should also be refactored to use the same core implementation as `box.Exec` and `RunScript` (which is your TODO above)

Reply:
Hmmm ok I'm not too familiar with `box.Exec`, I can look into it. I'm new to this code, so this type of feedback is really useful.